week2/main.js

```diff
@@ -0,0 +1,130 @@
+class ListItem extends React.Component {
+  render() {
+    const { description, deadline } = this.props;
+    return (
+      <div>
+        {description}, {deadline}
+      </div>
+    );
+  }
+}
+
+class DynamicList extends React.Component {
+  render() {
+    return (
+      <div>
+        {this.props.todos.map(todo => {
+          return (
+            <div key={todo.id} className={`${todo.done}`}>
+              <ListItem description={todo.description} deadline={todo.deadline} />
+              <button
+                className="delete"
+                onClick={() => {
+                  this.props.handleDelete(todo.id);
+                }}
+              >
+                Delete
+              </button>
+            </div>
+          );
+        })}
+      </div>
+    );
+  }
+}
+
+class AddItem extends React.Component {
+  handleChange = e => {
+    this.setState({
+      value: e.target.value,
+    });
+  };
+  handleSubmit = e => {
+    e.preventDefault();
+    this.props.onSubmit(this.state.value);
+    e.target.taskInput.value = '';
```

Comment:
rather than mutating the dom property directly, how about using set state here again ?